### ISSUE TITLE
False positive after committing a suggestion

### ISSUE THREAD

**Issue by informeti** [mongolyy/reviewdog-action-biome#66]
I just tried out this action, and formatting and reporting both seem to work out-of-the-box. Excellent!

But when I committed the suggestion in GitHub, the changed lines were reported again as an error, except that the diff doesn't really show a difference.

It _seems_ to be due to a missing line at the end of the file, which Biome seems to unconditionally require... If my understanding is correct, this action, reviewdog or biome strips the empty line at the end of the file.

Truth be told, I'd rather have Biome accept this, but afaik Biome is a bit conservative on that side of things.

---

**Before committing the suggestion (so far, so good)**:
![image](https://github.com/user-attachments/assets/a852da9e-d7aa-418e-9366-bdf05280f118)

**After committing the suggestion (not good)**:
![image](https://github.com/user-attachments/assets/46ee97a6-ef74-42c2-b4ff-19d819e97e3a)

Workflow file:

```yaml
name: check

on:
  pull_request:

jobs:
  biome:
    runs-on: ubuntu-latest
    permissions:
      contents: read
      pull-requests: write
    steps:
      - uses: actions/checkout@v4
      - uses: mongolyy/reviewdog-action-biome@v1
        with:
          github_token: ${{ secrets.github_token }}
          reporter: github-pr-review
          fail_on_error: true
```

Log:

```
Run mongolyy/reviewdog-action-biome@v1
Run biomejs/setup-biome@v2.2.1
Run reviewdog/action-setup@v1.3.0
Run set -eu
🐶 Installing reviewdog ... https://github.com/reviewdog/reviewdog
Run set -eu
📖 reviewdog -h
Run $GITHUB_ACTION_PATH/script.sh
 Running Biome with reviewdog 🐶 ...
Run reviewdog/action-suggester@v1
Run set -euo pipefail
🐶 Installing reviewdog ... https://github.com/reviewdog/reviewdog
Run set -euo pipefail
📖 reviewdog -h
Run set -euo pipefail
  set -euo pipefail
  . "$GITHUB_ACTION_PATH/script.sh"
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    INPUT_GITHUB_TOKEN: ***
    INPUT_TOOL_NAME: Biome
    INPUT_LEVEL: error
    INPUT_FILTER_MODE: added
    INPUT_FAIL_ON_ERROR: true
    INPUT_REVIEWDOG_FLAGS: 
    INPUT_CLEANUP: true
Saved working directory and index state WIP on (no branch): 57a96d7 Merge 9fc052bdd7b9084[33](snip#step:3:39)30de0bbc456e6ccb9436770 into 560b9777d43c5fa6cc80342c6fd4fa50e878fe28
ops/main.ts:3:-export * as organization from "./organization/main";
ops/main.ts:3:+export * as organization from "./organization/main";
reviewdog: input data has violations
Error: Process completed with exit code 1.
```

---

PS: Thank you for this wonderful action!

**Comment by mongolyy** [comment]
Thank you for using action! 🤩 
I'm happy that you were pleased.

After committing the suggestion, please try checking it out.
Could you try running `biome check` or `biome ci` commands to see if there are any differences?

On the back end, if there are any differences when action run `biome check --write`, a commit suggestion will be made.
I'm wondering if there are any differences in things like line endings.

**Comment by mongolyy** [comment]
@informeti 
Thank you more information!

The problem has been reproduced at #67 .

I think the settings for the tool called [reviewdog](https://github.com/reviewdog/reviewdog) used in this action are a little suspicious.

I think https://github.com/reviewdog/reviewdog/issues/1580 is similar.
However, setting `filter_mode: diff_context` did not solve this problem at #67.

I will investigate this in more detail this weekend.

**Comment by mongolyy** [comment]
@informeti 
I did some more research.

I came to the conclusion that there might be a problem with the tool called reviewdog that this tool uses.

Therefore, I created an issue and wait for the response.
https://github.com/reviewdog/reviewdog/issues/1927

**Comment by mongolyy** [comment]
@informeti 

I’ve confirmed that the original problem ([reviewdog/reviewdog#1927](https://github.com/reviewdog/reviewdog/issues/1927)) has been resolved.
Now, comments about missing EOF new lines are properly posted as expected.

You can see this in action here: [#69](https://github.com/mongolyy/reviewdog-action-biome/pull/69)
<img width="856" height="487" alt="Image" src="https://github.com/user-attachments/assets/adee16ad-356f-4db4-aa46-de588dcb91fb" />

I close this issue. If you still encounter any problems, please feel free to leave a comment.

Thanks again for your patience!
